android/java/org/chromium/chrome/browser/app/domain/NetworkModel.java

```diff
@@ -129,19 +136,8 @@ public NetworkModel(BraveWalletService braveWalletService, JsonRpcService jsonRp
                 }
             }
         });
-        _mChainId.addSource(mSharedData.getCoinTypeLd(), coinType -> {
-            mJsonRpcService.getNetwork(coinType, mOrigin, networkInfo -> {
-                String chainId = networkInfo.chainId;
-                String id = BraveWalletConstants.MAINNET_CHAIN_ID;
-                if (TextUtils.isEmpty(chainId)) {
-                    mJsonRpcService.setNetwork(
-                            id, mSharedData.getCoinType(), mOrigin, hasSetNetwork -> {});
-                } else {
-                    id = chainId;
-                }
-                _mChainId.postValue(id);
-            });
-        });
+        _mChainId.addSource(mMode, mode -> { updateChainId(); });
```

Comment:
We don't need mode as a live data, just keeping a reference should work , `updateChainId()` should be called when `updateMode` is called.

Reply:
https://github.com/brave/brave-core/pull/19329/commits/2b62a3db430107196b574a0d05990d585d0ee963